### PR TITLE
Implement Luma3DS gdb hio

### PR DESCRIPTION
**Pull request by TuxSH**
For debugged application, requires upcoming release/latest commit

### CODE REVIEW

File: libctru/source/gdbhio.c [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+{
+	return _gdbHioSendSyncRequest("open", "siI", pathname, _gdbHioExportOpenFlags(flags), _gdbHioExportFileMode(mode));
+}
+
+int gdbHioClose(int fd)
+{
+	return _gdbHioSendSyncRequest("close", "i", fd);
+}
+
+int gdbHioRead(int fd, void *buf, unsigned int count)
```
Comment (mtheall) [inline libctru/source/gdbhio.c]: Why not pass through `count` as `unsigned long` and use `L`?

File: libctru/source/gdbhio.c [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+{
+	return _gdbHioSendSyncRequest("read", "ipI", fd, buf, count);
+}
+
+int gdbHioWrite(int fd, const void *buf, unsigned int count)
```
Comment (mtheall) [inline libctru/source/gdbhio.c]: Why not pass through `count` as `unsigned long` and use `L`?

File: libctru/source/gdbhio.c [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+{
+	return _gdbHioSendSyncRequest("write", "ipI", fd, buf, count);
+}
+
+int gdbHioLseek(int fd, off_t offset, int flag)
```
Comment (mtheall) [inline libctru/source/gdbhio.c]: `off_t` is signed; does that mean you should be using `l`?

File: libctru/source/gdbhio.c (hunk 1 above)
Comment (TuxSH) [reply libctru/source/gdbhio.c]: GDB uses unsigned int (32-bit) https://sourceware.org/gdb/onlinedocs/gdb/read.html#read

File: libctru/source/gdbhio.c (hunk 2 above)
Comment (TuxSH) [reply libctru/source/gdbhio.c]: Same as above

File: libctru/source/gdbhio_dev.c [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+int gdbHioDevGetStderr(void)
+{
+	return _gdbHioDevImportFd(GDBHIO_STDERR_FILENO);
+}
+
+int gdbHioDevRedirectStdStreams(bool in, bool out, bool err)
+{
+	int ret = 0;
+	if (in && (ret = dup2(gdbHioDevGetStdin(), STDIN_FILENO) < 0)) return ret;
```
Comment (WinterMute) [inline libctru/source/gdbhio_dev.c]: The file descriptors returned from gdbHioDevGetStd* need closed here so the user can close the new descriptors & undo the redirection without leaking resources. dup2 increases the refcount on the cloned handle so it won't be released when the first descriptor that references it is closed.

**Pull request merged**